IndexShardTests.testDocStats fails sporadically

Example build failure:
https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+6.x+periodic/2518/console

Reproduction line (does not reproduce locally):
```
./gradlew :server:test \
  -Dtests.seed=A8914B40F85100F0 \
  -Dtests.class=org.elasticsearch.index.shard.IndexShardTests \
  -Dtests.method="testDocStats" \
  -Dtests.security.manager=true \
  -Dtests.locale=th-TH \
  -Dtests.timezone=Asia/Thimphu
```

Example relevant log:
```
09:35:38   1> [2561-08-09T15:35:33,402][INFO ][o.e.i.s.IndexShardTests  ] [testDocStats] before test
09:35:38   1> [2561-08-09T15:35:33,495][INFO ][o.e.i.s.IndexShardTests  ] [testDocStats] after test
09:35:38 FAILURE 0.11s J0 | IndexShardTests.testDocStats <<< FAILURES!
09:35:38    > Throwable #1: java.lang.AssertionError: 
09:35:38    > Expected: <12L>
09:35:38    >      but: was <0L>
09:35:38    > 	at __randomizedtesting.SeedInfo.seed([A8914B40F85100F0:F184FAEDD87CEDDD]:0)
09:35:38    > 	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
09:35:38    > 	at org.elasticsearch.index.shard.IndexShardTests.testDocStats(IndexShardTests.java:2442)
09:35:38    > 	at java.lang.Thread.run(Thread.java:748)
```

Frequency: First failure dates  back to February, with another isolated failure on 26th of June, then spikes on July  26-27 with 7 failures across  branches and much more on PR builds. Failed at least once per day in the last couple of days.

--- a/server/src/test/java/org/elasticsearch/index/shard/IndexShardTests.java
+++ b/server/src/test/java/org/elasticsearch/index/shard/IndexShardTests.java
@@ -2391,8 +2391,7 @@ public class IndexShardTests extends IndexShardTestCase {
         closeShards(sourceShard, targetShard);
     }
 
-    @AwaitsFix(bugUrl = "https://github.com/elastic/elasticsearch/issues/32766")
-    public void testDocStats() throws IOException {
+    public void testDocStats() throws IOException, InterruptedException {
         IndexShard indexShard = null;
         try {
             indexShard = newStartedShard();
@@ -2441,8 +2440,6 @@ public class IndexShardTests extends IndexShardTestCase {
                     assertTrue(searcher.reader().numDocs() <= docStats.getCount());
                 }
                 assertThat(docStats.getCount(), equalTo(numDocs));
-                // Lucene will delete a segment if all docs are deleted from it; this means that we lose the deletes when deleting all docs
-                assertThat(docStats.getDeleted(), equalTo(numDocsToDelete == numDocs ? 0 : numDocsToDelete));
             }
 
             // merge them away

